lib/api-versions/version_check.rb

```
@@ -23,11 +23,27 @@ def accepts_proper_format?(accept_string)
     end
 
     def matches_version?(accept_string)
-      accept_string =~ /version\s*?=\s*?#{@process_version}\b/
+      process_version = Gem::Requirement.new(@process_version)
+      begin
+        # Ignore anything past the minor version
+        # Versions can contain lowercase letters to indicate prerelease: http://ruby-doc.org/stdlib-2.0/libdoc/rubygems/rdoc/Gem/Version.html
+        accept_version = Gem::Version.new version_regex('(?<version>[0-9]+.?[0-9a-z]*)').match(accept_string).try(:[], :version)
+        process_version.satisfied_by? accept_version
+      rescue
```

The thinking here is that `accept_string` is user input, so rather than allow 500's to surface and leak information about the implementation we will silently ignore the given version.
Would you prefer `rescue StandardError`?